### PR TITLE
Put the processing latency measurement into the IR

### PR DESCRIPTION
**Pull request by mgritter**
Calculate the latency from the information in the ParsedNetworkTraffic objects.

For HAR replay, rebuild ParsedNetworkTraffic timestamps so that they will give
the same latency result (as is already present in the HAR's Timings.Wait. This is sort of gross, but
there didn't seem to be a good way to get the information from point A to point B without more
work than it was worth.

The original code had comments suggesting that type incompatibility prevented re-use 
of the Martian library's HAR parsing; I'm guessing that is because it originally used ints?

### CODE REVIEW

**Comment by mgritter** [comment]
Marked as draft because it has to go in *after* changes to the server side, but tested and appears to work.

**Review by thatplguy — APPROVED** [review]
Looks great!

File: apispec/har.go [hunk 1]
```diff
@@ -39,6 +41,67 @@ func processHAR(inboundCol, outboundCol trace.Collector, p string) error {
...
+	ResponseStart time.Time
+	ResponseEnd   time.Time
+}
+
+func advanceTime(current time.Time, ms *float32) time.Time {
+	if ms == nil || *ms <= 0.0 {
+		return current
+	}
+	// convert ms to nanoseconds
+	duration := time.Duration(int64(*ms * 1_000_000.0))
```
Comment (liujed) [inline apispec/har.go:65]: How about `duration := *ms * time.Millisecond`?

File: trace/backend_collector.go [hunk 2]
```diff
@@ -73,6 +76,39 @@ func (r witnessWithInfo) toReport(dir kgxapi.NetworkDirection) (*kgxapi.WitnessR
...
+
+	// Missing data, leave as default value in protobuf
+	if requestEnd.IsZero() || responseStart.IsZero() {
+		return
+	}
+
+	// HTTPMethodMetadata only for now
+	if meta := spec_util.HTTPMetaFromMethod(w.witness.Method); meta != nil {
+		latency := responseStart.Sub(requestEnd)
+		meta.ProcessingLatency = float32(latency.Microseconds()) / 1000.0
```
Comment (liujed) [inline trace/backend_collector.go:108]: How about `meta.ProcessingLatency = float32(latency/time.Millisecond)`?

**Review by liujed — APPROVED** [review]

File: apispec/har.go (hunk 1 above)
Comment (mgritter) [reply apispec/har.go:65]: That's where I started, it doesn't work.  Only constants can do that.  :(

https://play.golang.org/p/f71AyeNmJoG

./prog.go:9:18: invalid operation: *ms * time.Millisecond (mismatched types float32 and time.Duration)

File: trace/backend_collector.go (hunk 2 above)
Comment (mgritter) [reply trace/backend_collector.go:108]: Same problem, can't divide a float by a Duration, have to convert it to a Duration first.  (And the time you get is then a Duration, which is an int64, so the fractional part would be gone.)

Comment (liujed) [reply trace/backend_collector.go:108]: `latency / time.Millisecond` does type-check, but you're right that you lose the fractional part. https://play.golang.org/p/kTyRO1PFp0b

File: apispec/har.go (hunk 1 above)
Comment (liujed) [reply apispec/har.go:65]: Alas, how disappointing.

**Pull request merged**